src/drizzleReactHooks.js

```diff
@@ -0,0 +1,203 @@
+import React, {
+  createContext,
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState
+} from 'react'
+import PropTypes from 'prop-types'
+import debounce from 'debounce'
+import shallowequal from 'shallowequal'
+
+const Context = createContext()
+export const useDrizzle = () => useContext(Context)
+export const useDrizzleState = (mapState, args) => {
+  const { drizzle } = useDrizzle()
+  const mapStateRef = useRef(mapState)
+  mapStateRef.current = mapState
+  const argsRef = useRef(args)
+  const [state, setState] = useState(
+    mapStateRef.current(drizzle.store.getState())
+  )
+  const stateRef = useRef(state)
+  if (!shallowequal(argsRef.current, args)) {
+    argsRef.current = args
+    const newState = mapStateRef.current(drizzle.store.getState())
+    if (!shallowequal(stateRef.current, newState)) {
+      stateRef.current = newState
+      setState(newState)
+    }
+  }
+  useEffect(
+    () => {
+      const debouncedHandler = debounce(() => {
+        const newState = mapStateRef.current(drizzle.store.getState())
+        if (!shallowequal(stateRef.current, newState)) {
+          stateRef.current = newState
+          setState(newState)
+        }
+      })
+      const unsubscribe = drizzle.store.subscribe(debouncedHandler)
+      return () => {
+        unsubscribe()
+        debouncedHandler.clear()
+      }
+    },
+    [drizzle.store]
```

Comment:
Does this just get re-rendered every time anyway since it only does a shallow comparison?

Reply:
Isn't the redux store mutable? I.e. `store.getState()` changes, but not `store`.

Reply:
Hmm, I'm really not sure, I'm just curious. I'm happy leaving this in.

Reply:
Yeah, it's a literal that just gets returned here:
https://github.com/reduxjs/redux/blob/f0ee46bd407c08cd0991e84f22b5a28ed75e5149/src/createStore.js#L286-L292